sklearn/utils/estimator_checks.py

```diff
@@ -1059,7 +1063,8 @@ def _check_transformer(name, transformer_orig, X, y):
             assert _num_samples(X_pred3) == n_samples
 
         # raises error on malformed input for transform
-        if hasattr(X, 'T') and not _safe_tags(transformer, "stateless"):
+        if hasattr(X, 'T') and not _safe_tags(transformer, "stateless") \
```

Comment:
The fix I've used in other places in this file (I think) is to slice X instead of using ``X.T``. We just want something with different shape, so if you pass ``X[:, ::-1]`` instead of ``X.T`` the test works for all cases.
Otherwise you're now skipping this test for pairwise estimators.

Reply:
Thanks for the suggestion.  Would it work to use `X[:-1, :-1]` instead or does that risk messing up the test for other estimators?  I am thinking that `X[:, :-1]` will always raise a value error just because the input is not square, not because the dimensions of the input are different.

Reply:
Thanks for the suggestions.  One issue I am thinking is that for pairwise estimators, a ValueError could be triggered on `X[:, :-1]` because the input would be non-square instead of different number of dimensions.  So if X was 10x10, the estimator would throw an error on 10x9 input but possibly not 9x9 input.  Is this a problem? 

Would it make sense to do `pairwise_estimator_convert_X(X[:-1, :]).T`?  This would take a 10x10 array X to 10x9 for most estimators and 9x9 for pairwise estimators.

Reply:
Pairwise estimators should allow rectangular data in ``transform`` or ``predict``. I think they all do, the number of features should be the number of rows==columns in the training data.

Reply:
It doesn't look to me as if ``KernelCenterer`` requires square data in ``transform``.